sirepo/raydata/scan_monitor.py

```diff
@@ -171,6 +162,38 @@ def statuses_for_scans(cls, catalog_name, rduids):
             .all()
         )
 
+    @classmethod
+    def _db_upgrade(cls):
+        def _fixup_running_statuses():
+            for x in cls.session.query(cls).filter(
+                cls.status.in_(_AnalysisStatus.NON_STOPPED)
+            ):
+
+                cls.set_scan_status(
+                    PKDict(rduid=x.rduid, catalog_name=x.catalog_name),
+                    _AnalysisStatus.ERROR,
+                )
+
+        def _rduid_in_schema():
```

Comment:
Is there a reason to prefer this over the code I shared? The code I shared I grabbed from how we are looking up columns elsewhere in sirepo https://github.com/radiasoft/sirepo/blob/c12aedc0d2d6b60186015dc88091baafb2698503/sirepo/auth_db/__init__.py#L41.

Reply:
The code you shared will always be true because it checks the schema defined in the class. we need to check the actual db schema

Reply:
Aha, thanks!